### PR TITLE
Submit Custom html card code in github

### PR DESCRIPTION
**Pull request by praveena-pappu**

### CODE REVIEW

File: viz/CustomHTMLCard/src/manifest.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "Custom HTML Card",
+  "organization": "Data Studio",
+  "description": "Community Viz lets users design and render their own viz in Data Studio. Using this card generator Community Viz we will let users render individual cards for each of their data rows. User can provide their own style. Sample format body{background-color:#aebecf} .row{float:left;width:30%;margin:0.5em;border:0.05em solid black;}",
+  "logoUrl": "https://storage.googleapis.com/lego-devrel-testing/CustomHTMLCard/HTMLCard.png",
+  "supportUrl": "https://issuetracker.google.com//issues/new?component=801069&template=1385925",
+  "privacyPolicyUrl": "https://policies.google.com/privacy",
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/manifest.json]: I don't think you can use the Google privacy policy and ToS. Delete these for now, I'll check on what should be used.

File: viz/CustomHTMLCard/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,4425 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (y3l2n) [inline viz/CustomHTMLCard/yarn.lock]: There shouldn't be both yarn.lock and package.lock - if you're using npm, keep package.lock if you're using yarn, use yarn.lock

**Review by musiciancodes — CHANGES_REQUESTED** [review]

File: viz/CustomHTMLCard/README.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Data Studio Community Visualization code to create custom HTML cards
+
+Data Studio [community visualizations][community viz] allow you to write custom
```
Comment (y3l2n) [inline viz/CustomHTMLCard/README.md]: Community Visualizations is usually capitalized. Markdown links need a link to work properly. Look up reference links here: https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

File: viz/CustomHTMLCard/README.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+JavaScript visualizations for [Google Data Studio][datastudio].
+
+## About this Community Visualization
+
+This visualization helps user to create custom HTML cards using the row data given as input by the user. 
+Each cell from the row is read and the whole row is shown as HTML card to the user. User can customize the HTML cards by providing 
+styling in the "Style" tab. User can mention the styling for the HTML body and row using the tags and class names. The classname for the row data is ".row". Sample styling looks like: "body{background-color:#aebecf} .row{float:left;width:30%;margin:0.5em;border:0.05em solid black;}"
```
Comment (y3l2n) [inline viz/CustomHTMLCard/README.md]: The sample styling belongs in a "code block"

Should also link to the sample report here and maybe have screenshots of how/where to configure it

File: viz/CustomHTMLCard/src/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    }
+    function loadData() {
+      $.each(nameData, function(i, row) {
+        fieldData = row.name;
+        fieldArray.push(fieldData);
+      });
+      $.each(rowData, function(j, row) {
+        dimensionData = row['chartDimension'];
+        var rowDiv = $("<div class='row'>");
+        for (var k in dimensionData) {
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/index.js]: const k

File: viz/CustomHTMLCard/src/index.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+          let finalData = $('<div class="field">');
+          let fieldElement =
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/index.js]: Any reason why this can't be a const?

File: viz/CustomHTMLCard/src/index.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      loadData();
+    }
+    function loadData() {
+      $.each(nameData, function(i, row) {
+        fieldData = row.name;
+        fieldArray.push(fieldData);
+      });
+      $.each(rowData, function(j, row) {
+        dimensionData = row['chartDimension'];
+        var rowDiv = $("<div class='row'>");
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/index.js]: should either be let or const

File: viz/CustomHTMLCard/src/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+          finalData.append(fieldElement + data);
+          rowDiv.append(finalData);
+        }
+        mainDiv.append(rowDiv);
+      });
+      $('body')
+        .addClass('container')
+        .append(mainDiv);
+    }
+    if (style != undefined && style != '') {
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/index.js]: use !== here

File: viz/CustomHTMLCard/src/manifest.json [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  "termsOfServiceUrl": "https://support.google.com/datastudio/answer/7019158",
+  "packageUrl": "https://datastudio.google.com/open/1QhuGDiHKUQ-yBs4nfAOesdhr9RBeQPvm",
```
[the remark this answers is not in the document]
Comment (y3l2n) [reply viz/CustomHTMLCard/src/manifest.json]: The report should look a little more polished - more whitespace, less crowded. Take a look at the gallery reports

File: viz/CustomHTMLCard/src/manifest.json [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "Custom HTML Card",
+  "organization": "Data Studio",
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/manifest.json]: Organization should be "Data Studio Developer Relations"

File: viz/CustomHTMLCard/src/manifest.json [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+  "description": "Community Viz lets users design and render their own viz in Data Studio. Using this card generator Community Viz we will let users render individual cards for each of their data rows.",
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/manifest.json]: Description should be shorter: closer to 10-15 words max

**Review by musiciancodes — CHANGES_REQUESTED** [review]

File: viz/CustomHTMLCard/.project [hunk 12]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/.project]: removed the file

File: viz/CustomHTMLCard/.settings/org.eclipse.wst.jsdt.ui.superType.container [hunk 13]
```diff
@@ -0,0 +1 @@
+org.eclipse.wst.jsdt.launching.JRE_CONTAINER
```
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/.settings/org.eclipse.wst.jsdt.ui.superType.container]: removed the file

File: viz/CustomHTMLCard/README.md [hunk 14]
```diff
@@ -0,0 +1,119 @@
+# Data Studio Community Visualization Local Development Template
```
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/README.md]: Updated readme

File: viz/CustomHTMLCard/src/localMessage.js [hunk 15]
```diff
@@ -0,0 +1,51 @@
+/**
```
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/localMessage.js]: removed the file

File: viz/CustomHTMLCard/src/index.js [hunk 16]
```diff
@@ -0,0 +1,72 @@
+const dscc = require('@google/dscc');
+const $ = require('jquery');
+
+export const LOCAL = false;
+
+function drawViz(data) {
+  var nameData = data.fields.chartDimension;
```
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/index.js]: updated the var to let

File: viz/CustomHTMLCard/src/manifest.json (hunk 1 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/manifest.json]: removed PP and TOS links

File: viz/CustomHTMLCard/yarn.lock (hunk 2 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/yarn.lock]: removed yan.lock

File: viz/CustomHTMLCard/README.md (hunk 4 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/README.md]: removed the code block and updated the link from Viz report

File: viz/CustomHTMLCard/src/index.js (hunk 5 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/index.js]: updated

File: viz/CustomHTMLCard/src/index.js (hunk 6 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/index.js]: updated to const

File: viz/CustomHTMLCard/src/index.js (hunk 7 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/index.js]: updated to const

File: viz/CustomHTMLCard/src/index.js (hunk 8 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/index.js]: updated as per review comment

File: viz/CustomHTMLCard/src/manifest.json (hunk 10 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/manifest.json]: Updated

File: viz/CustomHTMLCard/src/manifest.json (hunk 11 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/manifest.json]: changed the description

File: viz/CustomHTMLCard/src/manifest.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (y3l2n) [reply viz/CustomHTMLCard/src/manifest.json]: The first page of the report should make it clearer what I'm looking at.

File: viz/CustomHTMLCard/src/manifest.json [hunk 17]
```diff
@@ -0,0 +1,25 @@
...
+  "supportUrl": "https://issuetracker.google.com//issues/new?component=801069&template=1385925",
+  "privacyPolicyUrl": "",
+  "termsOfServiceUrl": "",
+  "packageUrl": "https://datastudio.google.com/open/1QhuGDiHKUQ-yBs4nfAOesdhr9RBeQPvm",
+  "organizationUrl":"",
+  "devMode": false,
+  "components": [
+    {
+      "id": "htmlCard",
+      "name": "CustomHTMLCard",
```
Comment (y3l2n) [inline viz/CustomHTMLCard/src/manifest.json]: "HTML Card Template"

File: viz/CustomHTMLCard/README.md [hunk 18]
```diff
@@ -0,0 +1,11 @@
+# Data Studio Community Visualization code to create custom HTML cards
+
+Data Studio [community visualizations][community viz] allow you to write custom
+JavaScript visualizations for [Google Data Studio][datastudio].
+
+## About this Community Visualization
+
+This visualization helps user to create custom HTML cards using the row data given as input by the user. 
+Each cell from the row is read and the whole row is shown as HTML card to the user. User can customize the HTML cards by providing 
+styling in the "Style" tab. User can mention the styling for the HTML body and row using the tags and class names. The classname for the row data is ".row". Sample styling looks like: https://datastudio.google.com/c/u/0/reporting/1QhuGDiHKUQ-yBs4nfAOesdhr9RBeQPvm/page/NwtEB
```
Comment (y3l2n) [inline viz/CustomHTMLCard/README.md]: use a reference link here.

File: viz/CustomHTMLCard/README.md (hunk 3 above)
Comment (y3l2n) [reply viz/CustomHTMLCard/README.md]: Add a reference link for community viz

**Review by musiciancodes — CHANGES_REQUESTED** [review]

File: viz/CustomHTMLCard/README.md (hunk 3 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/README.md]: Updated as per review comments

File: viz/CustomHTMLCard/src/manifest.json (hunk 9 above)
[the remark this answers is not in the document]
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/manifest.json]: Updated

File: viz/CustomHTMLCard/src/manifest.json (hunk 17 above)
Comment (praveena-pappu) [reply viz/CustomHTMLCard/src/manifest.json]: Updated the name

**Review by musiciancodes — APPROVED** [review]

**Pull request merged**